cli/flox-rust-sdk/src/models/env_registry.rs

```diff
@@ -426,6 +434,29 @@ pub fn deregister_activation(
     Ok(current_activations)
 }
 
+/// Returns whether this watchdog has been started without a need for it
+/// e.g. another watchdog is already monitoring this activation.
+pub fn should_bail_at_startup(reg: &EnvRegistry, path_hash: &str) -> bool {
+    reg.entry_for_hash(path_hash)
+        .map(|entry| !entry.activations.is_empty())
```

Comment:
Should we remove PIDs that are no longer running before checking that it's empty so that a dead activation doesn't prevent a watchdog from ever being started for an environment?

Reply:
It may be worth doing a best-effort pass over the PIDs, but since kqueue can't tell us about PIDs that have _already_ exited, you'll have to fall back to `kill -0` which we know has some holes. On Linux you could just use `/proc`.

Reply:
Implemented this in https://github.com/flox/flox/pull/2192/commits/82b5ca6d5972ffab9c7e6b4f904450984e3e0f2a